### DISCUSSION TITLE
Separate power and hvac_mode DPs on an air conditioner

### DISCUSSION THREAD

**Discussion by ilmakangas** [make-all/tuya-local#1843] [Q&A] [answered]
Hello

I have a YFA-05 air conditioner. It has separate DPs for power (ID 1, boolean) and for HVAC modes (ID 4 which is a string, either COOL, DRY or FAN). 

I have the following config

> dps:
>     - id: 1
>       name: hvac_mode
>       type: boolean
>       mapping:
>         - dps_val: false
>           value: "off"
>         - dps_val: true
>           constraint: mode
>           conditions:
>             - dps_val: COOL
>               value: cool
>             - dps_val: FAN
>               value: fan_only
>             - dps_val: DRY
>               value: dry
>     - id: 4
>       name: mode
>       type: string
>       hidden: true

The problem is that I cannot change the mode of the device. I debugged the device and it seems that if it's already running and you want to change the mode, you have to only send the command to change the ID 4 DP without commanding ID 1 = true at the same time. But as Tuya local seems to set both ID 4 = new mode and ID 1 = true at same time and it doesn't work.

Is it possible to fix the config to avoid resending the ID 1 value each time?

**Comment by make-all** [comment, marked as answer]
You can make the power a separate switch entity, and have a climate entity that does not support turn_on/turn_off services.

**Comment by krie** [comment]
Hello @make-all

I wanted to bring this up again, because I personally don't really like the idea of having a separate power switch entity.
Interacting with the device in HA dashboards (via more-info dialog or thermostat card) would be much easier, if there was an "OFF" HVAC mode provided for the climate entity, instead of a completely separate switch. In addition it is quite confusing to the user, that the UI always displays the last active HVAC mode, even though the device is powered off.

My device - a Klarstein Iceblock Ecosmart AC (https://github.com/make-all/tuya-local/pull/2001) - was showing the exact same behavior as described above, when trying to map the power DP into hvac_mode.

**Is there any chance, to fully support devices like these?** Thank you!

<img width="552" alt="image" src="https://github.com/user-attachments/assets/25e29307-06fd-429d-89f1-4e6c55dbc775">

**Reply by make-all** [reply]
If the device does not support setting two dps at once, then the native behaviour of HA cannot be supported, as the on/off needs to be seperate from mode setting. I have no plans to introduce complex stateful command sequencing to work around this.
